packages/astro/src/astro-components/unstyled/SignInButton.astro

```diff
@@ -0,0 +1,49 @@
+---
+import type { HTMLTag, Polymorphic } from 'astro/types'
+import type { SignInProps } from "@clerk/types";
+type Props<Tag extends HTMLTag> = Polymorphic<SignInProps & { as: Tag; id?: string; mode?: 'redirect' | 'modal' }>
+
+import { customAlphabet, urlAlphabet } from "nanoid";
+
+const safeId = customAlphabet(urlAlphabet, 10)();
```

Comment:
Starting to notice we're repeating this piece of logic in all our Astro components. I'll make a ticket that will added this